modules/apps/commerce/commerce-product-subscription-type-web/src/main/java/com/liferay/commerce/product/subscription/type/web/internal/MonthlyCPSubscriptionTypeImpl.java

```diff
@@ -142,6 +146,97 @@ else if (monthlyMode ==
 		return calendar.getTime();
 	}
 
+	@Override
+	public UnicodeProperties validateDeliverySubscriptionTypeSettingsProperties(
+			UnicodeProperties subscriptionTypeSettingsUnicodeProperties)
+		throws PortalException {
+
+		if (subscriptionTypeSettingsUnicodeProperties == null) {
+			return null;
+		}
+
+		int deliveryMonthlyMode = GetterUtil.getInteger(
+			subscriptionTypeSettingsUnicodeProperties.get(
+				"deliveryMonthlyMode"));
+
+		if ((deliveryMonthlyMode <
+				CPSubscriptionTypeConstants.MODE_ORDER_DATE) ||
+			(deliveryMonthlyMode >
+				CPSubscriptionTypeConstants.MODE_LAST_DAY_OF_MONTH)) {
+
+			throw new CPSubscriptionTypeSettingsException(
+				"Invalid deliveryMonthlyMode " + deliveryMonthlyMode);
+		}
+
+		HashMapBuilder.HashMapWrapper<String, String>
+			subscriptionTypeSettingsProperties = HashMapBuilder.put(
+				"deliveryMonthlyMode", String.valueOf(deliveryMonthlyMode));
+
+		if (deliveryMonthlyMode ==
+				CPSubscriptionTypeConstants.MODE_EXACT_DAY_OF_MONTH) {
+
+			int deliveryMonthDay = GetterUtil.getInteger(
+				subscriptionTypeSettingsUnicodeProperties.get(
+					"deliveryMonthDay"));
+
+			if ((deliveryMonthDay < 1) || (deliveryMonthDay > 31)) {
+				throw new CPSubscriptionTypeSettingsException(
+					"Invalid deliveryMonthDay " + deliveryMonthDay);
+			}
+
+			subscriptionTypeSettingsProperties.put(
+				"deliveryMonthDay", String.valueOf(deliveryMonthDay));
+		}
+
+		return UnicodePropertiesBuilder.create(
+			subscriptionTypeSettingsProperties.build(), true
+		).build();
```

Comment:
Make this a private method and pass "deliveryMonthlyMode" and "deliveryMonthDay" as params.
Call them monthlyModeParam and monthDayParam